### PR TITLE
BVAL-561

### PR DESCRIPTION
**Pull request by gunnarmorling**
Supersedes #134.

### CODE REVIEW

File: sources/constraint-definition.asciidoc [hunk 1]
```diff
@@ -455,7 +460,18 @@ if [classname]`@FrenchZipCode` is defined as
...
+[tck-testable]#If a constraint is used more than once as a composing constraint, the multi value constraints model as described in <<constraintsdefinitionimplementation-multipleconstraints>> is used.#
+
+To select a specific composing constraint, [methodname]`OverridesAttribute.constraintIndex` is used.
+[tck-testable]#If the composing constraints are directly given on the composing constraint, `constraintIndex` refers to the left-to-right order of the constraints of this type in which they are given on the composing constraint.#
+[tck-testable]#If the composing constraints are specified using their corresponding `List` annotation, `constraintIndex` refers to the index within the `value` array.#
+[tck-testable]
+--
+A composing constraint must not be given directly on the composing constraint and using the corresponding `List` annotation at the same time.
```
Comment (gsmet) [inline sources/constraint-definition.asciidoc]: Can we determine that?

Comment (kasabuta4) [inline sources/constraint-definition.asciidoc]: According to Java Language Specification 8 Section 9.7.5, applying repeatable annotations and their container annotation at the same time is compile-time error.
So, we can determine that.

https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.7.5

Comment (gunnarmorling) [inline sources/constraint-definition.asciidoc]: Actually it is allowed to have one instance of the repeatable annotation type and one instance of the container annotation. See the second example in the section you linked:

> This rule is designed to allow the following code:
>
> @Foo(1) @FooContainer({@Foo(2)})
> class A {}

But we can determine that as `AnnotatedElement#getDeclaredAnnotations()` returns the _directly present_ annotations, e.g. `@Pattern` and `@Pattern.List`. In this case an exception would have to be thrown.

File: spec-examples/src/test/java/org/beanvalidation/specexamples/constraintdefinition/constraintcomposition/overriding/constraintindex/EmmanuelsEmail.java [hunk 2]
```diff
@@ -34,12 +34,10 @@
...
 //tag::include[]
-@Pattern.List({
-		@Pattern(regexp = "[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,4}"), // email
-		@Pattern(regexp = ".*?emmanuel.*?") // emmanuel
-})
-@Constraint(validatedBy = {})
 @Documented
+@Constraint(validatedBy = {})
+@Pattern(regexp = "[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,4}"), // email
+@Pattern(regexp = ".*?emmanuel.*?") // emmanuel
```
Comment (kasabuta4) [inline spec-examples/src/test/java/org/beanvalidation/specexamples/constraintdefinition/constraintcomposition/overriding/constraintindex/EmmanuelsEmail.java]: remove a comma(,) between `@Pattern` annotations.

**Comment by kasabuta4** [comment]
Thanks for your revise.
I have checked it out, and it seems all right to me,
except for removing a comma(,) between `@Pattern` annotation in EmmanuelsEmail.java.

**Comment by gunnarmorling** [comment]
Thanks, good catch! Seems I didn't compile the example :)

**Comment by gunnarmorling** [comment]
Fixed the example. Thanks again, @kasabuta4!

**Pull request closed without merging**